packages/api/src/codegen/languages/typescript/index.ts

```diff
@@ -458,25 +392,117 @@ sdk.server('https://eu.api.example.com/v14');`),
+
+    const config: TsConfigJson = {
+      compilerOptions: {
+        checkJs: true,
+        esModuleInterop: true,
+        module: 'NodeNext',
+        resolveJsonModule: true,
+      },
+      include: ['./src/**/*'],
+      exclude: ['dist'],
+    };
+
+    sourceFile.addStatements(JSON.stringify(config, null, 2));
+
+    return sourceFile;
+  }
+
+  /**
+   * Create the `package.json` file that will ultimately make this SDK available to use.
+   *
+   */
+  createPackageJSON() {
+    const sourceFile = this.project.createSourceFile('package.json', '');
+
+    const hasTypes = !!Object.keys(this.schemas).length;
+
+    const info = this.spec.getDefinition().info;
+    let pkgVersion = semver.coerce(info.version);
+    if (!pkgVersion) {
+      // If the version that's in `info.version` isn't compatible with semver NPM won't be able to
+      // handle it properly so we need to fallback to something it can.
+      pkgVersion = semver.coerce('0.0.0') as SemVer;
+    }
+
+    const tsupOptions: Options = {
+      cjsInterop: true,
+      clean: true,
+      dts: true,
+      entry: [
+        './src/index.ts',
+        // If this SDK has schemas and generated types then we should also export those too so
+        // they're available to use.
+        hasTypes ? './src/types.ts' : '',
+      ].filter(Boolean),
+      // TODO: figure this out
+      // external: ['@readme/api-core'],
+      format: ['esm', 'cjs'],
+      minify: false,
+      shims: true,
+      sourcemap: true,
+      splitting: true,
+    };
+
+    const dependencies = Object.entries(this.requiredPackages)
+      .map(([dep, { version }]) => ({ [dep]: version }))
+      .reduce((prev, next) => Object.assign(prev, next));
+
+    const pkg: PackageJson = {
+      name: `@api/${this.identifier}`,
+      version: pkgVersion.version,
+      main: './dist/index.js',
+      types: './dist/index.d.ts',
+      module: './dist/index.mts',
+      exports: {
+        '.': {
+          import: './dist/index.mjs',
+          require: './dist/index.js',
+        },
+        ...(hasTypes
+          ? {
+              './types': {
+                import: './dist/types.d.mts',
+                require: './dist/types.d.ts',
+              },
+            }
+          : {}),
+      },
+      dependencies,
+      tsup: tsupOptions as JsonObject,
+    };
+
+    sourceFile.addStatements(JSON.stringify(pkg, null, 2));
+
+    return sourceFile;
+  }
+
   /**
    * Create our main schemas file. This is where all of the JSON Schema that our TypeScript typing
    * infrastructure sources its data from. Without this there are no types.
    *
    */
-  private createSchemasFile() {
-    const sourceFile = this.project.createSourceFile('schemas.ts', '');
-    const schemasDir = this.project.createDirectory('schemas');
+  private createSchemasFile(sourceDirectory: Directory) {
+    const sourceFile = sourceDirectory.createSourceFile('schemas.ts', '');
```

Comment:
These `sourceDirectory` changes mean that these files we're creating here (`schemas.ts` and `schemas/`) are now being created in a `src/` directory.